### PR TITLE
Bugfix Preprocessor + training without preload as tensors

### PR DESCRIPTION
**Pull request by jhgoh**
#### Preprocessor:
- Bugfix: preprocessor 실행시 마지막 파일에서 이벤트를 스킵하는 문제가 있었음
- Feature: image를 normlize할 때 로딩 된 이미지에 대해서만 max를 취함. file split하는 경우 어떤 파일들이 로딩되어 있느냐에 따라 결과가 달라짐. 이 단계에서는 normalization을 하지 않고 training시에 model에서 하는 편이 나아보임.
- Feature: output을 directory로 입력하면 data_*.h5를 기본값으로 사용.
- Compression algorithm자체를 유저가 선택하도록 변경 - LZ4가 가장 빠르므로 이를 권장함. chunk size는 1이 적당해 보임.

#### Training
- Change behaviour: 초기 로딩 단계에서 파일의 모든 내용을 tensor로 변환해 in-memory로 처리하던 것을 hdf에서 on-demand로 로딩해 처리하는 방식으로 변경함 (input hdf파일이 압축되지 않아야 함 -> chunk 1인 LZ4면 괜찮음.)
- pytorch의 dataloader에서의 multiprocessing기능을 사용할 수 있게 함. 구현시 주의사항으로, class내에서 HDF파일이 열린채로 process fork나 spawn이 일어나는 경우 압축 해제 문제나 파일 접근 불가 메시지가 나타남. multiproc으로 class가 복사 된 이후에 파일을 열어야 하므로 __getitem__()에서 최초 파일 접근 시에 파일 핸들을 열어두는 방식으로 구현해야 함.
- Feature: image를 normlize할 때 로딩 된 이미지에 대해서만 max를 취함. file split하는 경우 어떤 파일들이 로딩되어 있느냐에 따라 결과가 달라짐. 이 단계에서는 normalization을 하지 않고 training시에 model에서 하는 편이 나아보임. 모델에서 "cat" 이 있으면 max값을 concatenate하고, "fixed"가 있으면 scale하지 않고 abs값을 사용하는 식으로 변경함. 기존의 default모델은 scale하고 cat하지 않는 경우와 비슷함.
- genevent 수를 고침.

### CODE REVIEW

**Comment by jhgoh** [comment]
2080ti 기준으로 224x224이미지 처리에 epoch당 15분 가량 소요됨.
